### PR TITLE
Ike v1 logging tx transforms 5455 v2

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5455

Describe changes:
- ikev1 : log fields from transaction instead of fields from state

ping @satta @fhonza as you have worked on ikev1

Replaces #7667 with S-V passing by keeping in the state the one transform chosen by the server

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 8424

**Comment by satta** [comment]
Looks OK at first glance. Now that it passes S-V, it should be fine for me. The reformats make it not obvious where the actual changes are.

**Comment by catenacyber** [comment]
> The reformats make it not obvious where the actual changes are.

Just check the second commit ;-)

**Comment by catenacyber** [comment]
https://github.com/OISF/suricata/pull/7668/commits

**Comment by catenacyber** [comment]
https://github.com/OISF/suricata/pull/7668/commits/84cb3ea66e5ede39d2a0d732212882c562b9c2df

File: rust/src/ike/ikev1.rs [hunk 1]
```diff
@@ -40,27 +40,27 @@ pub struct IkeV1Header {
 pub struct Ikev1ParticipantData {
     pub key_exchange: String,
     pub nonce: String,
-    pub vendor_ids: HashSet<String>,
-    /// nested Vec, outer Vec per Proposal/Transform, inner Vec has the list of attributes.
-    pub transforms: Vec<Vec<SaAttribute>>,
+    pub nb_transforms: usize,
```
Comment (victorjulien) [inline rust/src/ike/ikev1.rs:43]: could this be explicit type? usize depends on the architecture. Traffic properties should not.

Comment (catenacyber) [reply rust/src/ike/ikev1.rs:43]: Ok, I wanted to avoid the rust casts

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/7675

**Pull request closed without merging**